proposals/0003-spec.md

```diff
@@ -0,0 +1,249 @@
+
+# ﻿Healthcheck specification 1.0
+December 2016
+
+## Authors
+- Heiko Braun
+- Clement Escoffier
+- Heiko Rupp
+
+# Abstract
+This document defines a health check specification to be used by components that need to ensure a compatible wireformat, agreed upon semantics and possible forms of interactions between system components that need to determine the “liveliness” of computing nodes in a bigger system.
+
+## Guidelines
+
+Note that the force of these words is modified by the requirement level of the document in which they are used.
+
+1. MUST   This word, or the terms "REQUIRED" or "SHALL", mean that the
+  definition is an absolute requirement of the specification.
+
+2. MUST NOT   This phrase, or the phrase "SHALL NOT", mean that the
+  definition is an absolute prohibition of the specification.
+
+3. SHOULD   This word, or the adjective "RECOMMENDED", mean that there
+  may exist valid reasons in particular circumstances to ignore a
+  particular item, but the full implications must be understood and
+  carefully weighed before choosing a different course.
+
+4. SHOULD NOT   This phrase, or the phrase "NOT RECOMMENDED" mean that
+  there may exist valid reasons in particular circumstances when the
+  particular behavior is acceptable or even useful, but the full
+  implications should be understood and the case carefully weighed
+  before implementing any behavior described with this label.
+
+
+5. MAY – This word, or the adjective “OPTIONAL,” mean that an item is truly discretionary.
+Contributors
+* WildFly Swarm
+* Eclipse Vert.x
+* Hawkular
+Specification
+
+
+## Rationale
+
+The rationale for health checks in is to signal the state of a computing node to other machines (i.e. kubernetes service controller). It’s not intended (although could be used) as a monitoring solution for humans.
+
+## Goals
+* MUST be compatibility with Kubernetes health checks: http://kubernetes.io/docs/user-guide/liveness/
```

Comment:
K8s health check usage is consistent with Consul health check URLs, so it's a safe example/starting place.